### PR TITLE
Annotation changes only (part 3)

### PR DESCRIPTION
**Pull request by Maoni0**
+ moved the per heap/global fields to the last part of the gc_heap class so it's much easier to read them
+ made members that should be private actually private, so almost all of the fields are private now (instead of public)
+ changed the bookkeeping fields from MAINTAINED to INIT_ONLY to be more appropriate for regions
+ moved reset_mm_p into gc_heap class.
+ got rid of the frozen_object_p method - this shouldn't have been added, there's already an API on the GCHeap interface that provides this info
+ there are members that really don't need to be public but I'm trying to avoid too many kinds of changes at a time so I left them as public such as gc_started/wait_for_gc_done()/region_info enum. things like WaitLonger/stomp_write_barrier_ephemeral/reserved_memory_limit should really just be gc_heap methods

### CODE REVIEW

File: src/coreclr/gc/gcpriv.h [hunk 1]
```diff
@@ -131,7 +131,7 @@ inline void FATAL_GC_ERROR()
 // This means any empty regions can be freely used for any generation. For
 // Server GC we will balance regions between heaps.
 // For now disable regions for StandAlone GC, NativeAOT and MacOS builds
-#if defined (HOST_64BIT) && !defined (BUILD_AS_STANDALONE) && !defined(__APPLE__)
+#if defined (BUILD_AS_STANDALONE) && defined (HOST_64BIT) && !defined(__APPLE__)
```
Comment (PeterSolMS) [inline src/coreclr/gc/gcpriv.h:134]: This looks like a temporary change to flip clrgc.dll to regions...

File: src/coreclr/gc/gcpriv.h [hunk 2]
```diff
@@ -1323,7 +1327,7 @@ class mark_queue_t
 };
 
 //class definition of the internal class
-class gc_heap
+class gc_heap //
```
Comment (PeterSolMS) [inline src/coreclr/gc/gcpriv.h:1330]: There's a `//` starting a comment here, but no comment?

File: src/coreclr/gc/gcpriv.h (hunk 1 above)
Comment (Maoni0) [reply src/coreclr/gc/gcpriv.h:134]: ooh thanks! forgot to flip it back.

**Review by PeterSolMS — APPROVED** [review]
Looks good to me. Ended up using Windiff because it shows where lines got moved which made it easier to keep track of things.

**Pull request merged**